### PR TITLE
[cmd/telemetrygen] Add TraceID and SpanID support for logs in tracegen

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: cmd/telemetrygen/internal/logs/worker.go [hunk 1]
```diff
@@ -51,6 +55,24 @@ func (w worker) simulateLogs(res *resource.Resource, exporter exporter, telemetr
 		lattrs := log.Attributes()
 		lattrs.PutStr("app", "server")
 
+		if w.traceID != "" {
+			b, err := hex.DecodeString(w.traceID)
+			if err != nil {
+				w.logger.Fatal("failed to create traceID byte array from the given traceID, make sure the traceID is a hex representation of a [16]byte, like: 'ae87dadd90e9935a4bc9660628efd569'", zap.Error(err))
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/logs/worker.go:61]: Should we check for length as well?

Comment (jpkrohling) [reply cmd/telemetrygen/internal/logs/worker.go]: Yeah, I wasn't very happy with the missing test coverage for the error scenarios, so I added a Validate function as part of the Start, and will run this validation there. I added the extra length check as well.

File: cmd/telemetrygen/internal/logs/config.go [hunk 2]
```diff
@@ -28,4 +30,6 @@ func (c *Config) Flags(fs *pflag.FlagSet) {
 	fs.StringVar(&c.Body, "body", "the message", "Body of the log")
 	fs.StringVar(&c.SeverityText, "severity-text", "Info", "Severity text of the log")
 	fs.Int32Var(&c.SeverityNumber, "severity-number", 9, "Severity number of the log, range from 1 to 24 (inclusive)")
+	fs.StringVar(&c.TraceID, "trace-id", "", "TraceID of the log")
+	fs.StringVar(&c.SpanID, "span-id", "", "SpanID of the log")
```
Comment (mx-psi) [inline cmd/telemetrygen/internal/logs/config.go:34]: If you want to get even fancier, we could change the type of these, move the parsing logic to `UnmarshalText` and use https://pkg.go.dev/flag#TextVar

**Review by mx-psi — APPROVED** [review]

File: cmd/telemetrygen/internal/logs/config.go (hunk 2 above)
Comment (jpkrohling) [reply cmd/telemetrygen/internal/logs/config.go:34]: Apparently, this is not available in pflag, which is what we use here :-(

**Review by codeboten — APPROVED** [review]

File: cmd/telemetrygen/internal/logs/logs.go [hunk 3]
```diff
@@ -18,6 +19,13 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen/internal/common"
 )
 
+var (
+	errInvalidTraceIDLenght = fmt.Errorf("TraceID must be a 32 character hex string, like: 'ae87dadd90e9935a4bc9660628efd569'")
+	errInvalidSpanIDLenght  = fmt.Errorf("SpanID must be a 16 character hex string, like: '5828fa4960140870'")
+	errInvalidTraceID       = fmt.Errorf("failed to create traceID byte array from the given traceID, make sure the traceID is a hex representation of a [16]byte, like: 'ae87dadd90e9935a4bc9660628efd569'")
+	errInvalidSpanID        = fmt.Errorf("failed to create SpanID byte array from the given SpanID, make sure the SpanID is a hex representation of a [8]byte, like: '5828fa4960140870'")
```
Comment (bogdandrutu) [inline cmd/telemetrygen/internal/logs/logs.go:26]: When no format, use `errors.New`

File: cmd/telemetrygen/internal/logs/logs.go [hunk 4]
```diff
@@ -90,6 +105,32 @@ func Run(c *Config, exp exporter, logger *zap.Logger) error {
 	return nil
 }
 
+// Validate validates the test scenario parameters.
+func Validate(c *Config) error {
```
Comment (bogdandrutu) [inline cmd/telemetrygen/internal/logs/logs.go:109]: I think we will change to use confmap eventually, so better to follow that pattern:
```suggestion
func (c *Config) Validate error {
```

**Review by bogdandrutu — APPROVED** [review]

File: cmd/telemetrygen/internal/logs/logs.go (hunk 3 above)
Comment (jpkrohling) [reply cmd/telemetrygen/internal/logs/logs.go:26]: I used to think the same, but they are now equivalent and I find it more consistent to use only one for both cases.

File: cmd/telemetrygen/internal/logs/logs.go (hunk 4 above)
Comment (jpkrohling) [reply cmd/telemetrygen/internal/logs/logs.go]: good idea, I changed the code to reflect that

**Pull request merged**